servicebroker/test/routes/binding_test.js

```diff
@@ -0,0 +1,204 @@
+    .reply(500, {
+      'success': false,
+      'error': null,
+      'result': "error"
+    });
+}
+
+
+
+describe('binding RESTful API', function() {
+  var server, serviceInstanceId, serviceInstanceId2, orgId, spaceId, appId,appId2, bindingId;
+  serviceInstanceId = uuid.v4();
+  orgId = uuid.v4();
+  spaceId = uuid.v4();
+  appId = uuid.v4();
+  appId2 = uuid.v4();
+  bindingId = uuid.v4();
+  serviceInstanceId2 = uuid.v4();
+  var fake_service_condition = {
+    'serviceInstanceId': serviceInstanceId,
+    'orgId': orgId,
+    'spaceId': spaceId,
+    where: { 'serviceInstanceId': serviceInstanceId, 'orgId': orgId, 'spaceId': spaceId },
+  };
+  var policy = { "policy": "testPolicy" };
+
+  before(function(done) {
+    serviceInstance.sequelize.sync().then(function(result) {
+      binding.sequelize.sync().then(function(result) {
+        done();
+      }).catch(function(error1) {
+        console.log("Failed to sync model binding, error: " + error1);
+        done(error1);
+      });
+    }).catch(function(error2) {
+      console.log("Failed to sync model serviceInstance, error: " + error2);
+      done(error2);
+    });
+  });
+
+  beforeEach(function(done) {
+    delete require.cache[require.resolve('../../lib/index.js')];
+    server = require(path.join(__dirname, '../../lib/index.js'));
+    binding.truncate({ cascade: true }).then(function(result) {
+      serviceInstance.truncate({ cascade: true }).then(function(result) {
+        serviceInstance.findOrCreate(fake_service_condition).then(function(result) {
+          done();
+        }).catch(function(error1) {
+          done(error1);
+        });
+      }).catch(function(error2) {
+        done(error2);
+      });
+    }).catch(function(error3) {
+      done(error3);
+    });
+  });
+
+  afterEach(function(done) {
+    server.close(done);
+  });
+
+
+  context('Bind service ', function() {
+
+    context('when there is no policy in request', function() {
+      it("return a 400", function(done) {
+        supertest(server)
+          .put("/v2/service_instances/" + serviceInstanceId + "/service_bindings/" + bindingId)
+          .set("Authorization", "Basic " + auth)
+          .send({ "app_guid": appId })
+          .expect(400)
+          .expect('Content-Type', /json/)
+          .expect({"description":messageUtil.getMessage("POLICY_REQUIRED")}, done);
+      });
+    });
+
+    context('when there is no record', function() {
+      it("creates a new binding with 201", function(done) {
+        initNockBind201();
+        supertest(server)
+          .put("/v2/service_instances/" + serviceInstanceId + "/service_bindings/" + bindingId)
+          .set("Authorization", "Basic " + auth)
+          .send({ "app_guid": appId, "parameters": policy })
+          .expect(201)
+          .expect('Content-Type', /json/)
+          .expect({}, done);
+      });
+      context("when the api server return error", function() {
+        it("return a 400", function(done) {
```

Comment:
It returns the code just api returns to broker, there is a test case for code "500" .